app/src/main/java/com/example/yummy/ui/adapter/NewsAdapter.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.yummy.ui.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.yummy.R
+import com.example.yummy.data.model.News
+
+class NewsAdapter : RecyclerView.Adapter<NewsViewHolder>() {
+
+    var onItemClick: (News) -> Unit = { _ -> }
+    private var news = mutableListOf<News>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NewsViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_news, parent, false)
+        return NewsViewHolder(view, onItemClick)
+    }
+
+    override fun getItemCount() = news.size
+
+    override fun onBindViewHolder(holder: NewsViewHolder, position: Int) {
+        holder.bindData(news[position])
+    }
+
+    fun updateData(newList: List<News>) {
+        news.let {
```

Comment:
```suggestion
        news.run {
```
![scopefunction](https://user-images.githubusercontent.com/53286739/109445064-4e63f980-7a71-11eb-807d-1edade4a34ac.png)